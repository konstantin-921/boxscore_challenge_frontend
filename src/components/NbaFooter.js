import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'

const styles = theme => ({
  container: {
    textAlign: 'center',
    fontWeight: 'bold',
    height: '80px',
    color: '#ffffff',
    fontSize: '30px',
    lineHeight: '80px',
  },
  middleContainer: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'white',
    height: '80px',
    fontWeight: 'bold',
    backgroundColor: '#e0e0e0'
  }
});

class NbaFooter extends React.Component {
  render() {
    const { classes, data } = this.props
    const teamAway = data.away_team.last_name
    const teamHome = data.home_team.last_name
    const statusGeneral = data.event_information.status === 'completed' ? 'FINAL' : 'In progress'
    return (
      <div style={{ marginBottom: '100px' }}>
        <Grid container spacing={0} direction="row" alignItems="center" justify="center">
            <Grid item xs={4}>
              <div className={classes.container} style={{ backgroundColor: '#006400' }}>{teamAway}</div>
            </Grid>
            <Grid item xs={2}>
              <div className={classes.middleContainer}>{statusGeneral}</div>
            </Grid>
            <Grid item xs={4}>
              <div className={classes.container} style={{ backgroundColor: '#8B0000' }}>{teamHome}</div>
            </Grid>
        </Grid>
      </div>
    );
  }
}

NbaFooter.propTypes = {
  data: PropTypes.shape({
    away_team: PropTypes.shape({
      last_name: PropTypes.string.isRequired,
    }).isRequired,
    home_team: PropTypes.shape({
      last_name: PropTypes.string.isRequired,
    }).isRequired,
    event_information: PropTypes.shape({
      status: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default withStyles(styles)(NbaFooter);