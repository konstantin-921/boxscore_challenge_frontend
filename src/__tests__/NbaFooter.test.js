import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import NbaFooter from '../components/NbaFooter'

describe('NbaFooter component', () => {

  const props = {
    away_team: { last_name: 'JETS' },
    home_team: { last_name: 'PATRIOTS' },
    event_information: { status: 'completed' }
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<NbaFooter data={props} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})