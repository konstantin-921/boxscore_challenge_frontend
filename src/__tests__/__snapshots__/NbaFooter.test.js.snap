// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NbaFooter component renders correctly 1`] = `
<div
  style={
    Object {
      "marginBottom": "100px",
    }
  }
>
  <div
    className="MuiGrid-container-3 MuiGrid-align-items-xs-center-11 MuiGrid-justify-xs-center-20"
  >
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-4-35"
    >
      <div
        className="NbaFooter-container-1"
        style={
          Object {
            "backgroundColor": "#006400",
          }
        }
      >
        JETS
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-2-33"
    >
      <div
        className="NbaFooter-middleContainer-2"
      >
        FINAL
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-4-35"
    >
      <div
        className="NbaFooter-container-1"
        style={
          Object {
            "backgroundColor": "#8B0000",
          }
        }
      >
        PATRIOTS
      </div>
    </div>
  </div>
</div>
`;
